clients/cli/connections.c

```diff
@@ -2336,6 +2336,43 @@ typedef struct {
 	NMActiveConnection *active;
 } ActivateConnectionInfo;
 
+#if SYSTEMD_JOURNAL
+static void
+active_connection_hint (GString *return_text, ActivateConnectionInfo *info)
+{
+	NMRemoteConnection *connection;
+	GString *hint;
+	const GPtrArray *devices;
+	int i;
+
+	connection = nm_active_connection_get_connection (info->active);
+	g_return_if_fail (connection);
+
+	hint = g_string_new ("journalctl -xe ");
+	g_string_append_printf (hint, "NM_CONNECTION=%s",
+	                        nm_connection_get_uuid (NM_CONNECTION (connection)));
+
+	if (info->device) {
+		g_string_append_printf (hint, " + NM_DEVICE=%s", nm_device_get_iface (info->device));
+	} else {
+		devices = nm_active_connection_get_devices (info->active);
+		for (i = 0; i < devices->len; i++) {
+			g_string_append_printf (hint, " + NM_DEVICE=%s",
+			                        nm_device_get_iface (NM_DEVICE (g_ptr_array_index (devices, i))));
+		}
+	}
+
+	g_string_append (return_text, "\n");
+	g_string_append_printf (return_text, _("Hint: use '%s' to get more details."), hint->str);
+	g_string_free (hint, TRUE);
+}
+#else
```

Comment:
could you avoid the `#if #else` and instead move the compile time conditional inside the function?

Reply:
I prefer not to, the preprocessor macros intertwined with C code harm readability.